### PR TITLE
feat: publish and version

### PR DESCRIPTION
**Pull request by ethanve**

### CODE REVIEW

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -1,21 +1,78 @@
...
+      - name: Set Git Identity
+        run: |
+          git config --global user.name 'github-actions[bot]'
+          git config --global user.email 'github-actions[bot]@users.noreply.github.com'
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      - name: Install Dependencies
+        run: yarn
+      - name: Version
+        uses: ./.github/actions/tagger
```
Comment (jquense) [inline .github/workflows/publish.yml:74]: we don't need to push anything here? also do these tags point to the right commit? I thought the action context was a detached head?

Comment (ethanve) [reply .github/workflows/publish.yml:74]: Not with v2 of checkout! It actually checks out head I believe (or master specifically but that is what we want)

**Comment by ethanve** [comment]
I don't have write access so you're gonna have to merge this bad boy in

**Comment by ethanve** [comment]
@jquense we need an `NPM_TOKEN` secret to actually test publishing. Unless you don't want to pollute your npm namespace with these fake packages

**Comment by jquense** [comment]
go for it, this isn't published so you can use your own token

**Comment by ethanve** [comment]
@jquense don't have access to settings. You'll have to do it I think

**Pull request merged**